pallets/phala/src/stakepool.rs

```diff
@@ -904,9 +923,14 @@ pub mod pallet {
 			StakePools::<T>::insert(&pid, &pool_info);
 			Self::deposit_event(Event::<T>::Contribution {
 				pid,
-				user: who,
+				user: who.clone(),
 				amount: a,
 			});
+			Self::deposit_event(Event::<T>::ContributionShare {
```

Comment:
I suggest that instead of adding a new event, we can extend the existing event with the share number. If you only append a field to an event, it's always backward compatible.